File: app/src/main/java/com/example/elderapp/register/RegisterElderFragment.java
package com.example.elderapp.register;

import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.android.volley.AuthFailureError;
import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;
import com.example.elderapp.Global;
import com.example.elderapp.LoginActivity;
import com.example.elderapp.R;

import java.util.HashMap;
import java.util.Map;


public class RegisterElderFragment extends Fragment {

    private EditText etName, etPhone, etPassword, etPasswordC, etRemarks_ill, etRemarks_eating, etRemarks_other, etContactPhone, etContactEmail;
    private final String url = "https://www2.cs.ccu.edu.tw/~lwx109u/elderApp/register.php";
    private String username, phone, pass, passC, remarks_ill = "", remarks_eating = "", remarks_other = "", contactPhone, contactEmail;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View root = inflater.inflate(R.layout.fragment_register_elder, container, false);
        etName = root.findViewById(R.id.et_name);
        etPhone = root.findViewById(R.id.et_phone);
        etPassword = root.findViewById(R.id.et_password);
        etPasswordC = root.findViewById(R.id.et_passwordCheck);
        etRemarks_ill = root.findViewById(R.id.et_remarks1);
        etRemarks_eating = root.findViewById(R.id.et_remarks2);
        etRemarks_other = root.findViewById(R.id.et_remarks3);
        etContactPhone = root.findViewById(R.id.et_contactPhone);
        etContactEmail = root.findViewById(R.id.et_contactEmail);
        Button register = root.findViewById(R.id.btn_register);
        TextView toLogin = root.findViewById(R.id.tv_toLogin);

        register.setOnClickListener(view -> {
            username = etName.getText().toString().trim();
            phone = etPhone.getText().toString().trim();
            pass = etPassword.getText().toString().trim();
            passC = etPasswordC.getText().toString().trim();
            remarks_ill = etRemarks_ill.getText().toString().trim();
            remarks_eating = etRemarks_eating.getText().toString().trim();
            remarks_other = etRemarks_other.getText().toString().trim();
            contactPhone = etContactPhone.getText().toString().trim();
            contactEmail = etContactEmail.getText().toString().trim();

            if(!pass.equals(passC)){
                Global.putSnackBarR(etName, "密碼不相符");
            }
            else if(username.equals("") || phone.equals("") || contactPhone.equals("")){
                Global.putSnackBarR(etName, "請輸入完整資訊");
            }
            else {
                SQL();
            }

        });
        toLogin.setOnClickListener(view -> {
            startActivity(new Intent(getContext(), LoginActivity.class));
        });

        return root;
    }

    public void SQL() {
        StringRequest stringRequest = new StringRequest(Request.Method.POST, url, response -> {
            Log.d("connect", "Response: "+response);
            if (response.startsWith("success")) {
                Intent it = new Intent(getContext(), LoginActivity.class);
                it.putExtra("signUp", true);
                startActivity(it);
                getActivity().finish();
            }
            else if (response.startsWith("failure")) {
                Toast.makeText(getContext(), "註冊失敗", Toast.LENGTH_SHORT).show();
            }

        }, error -> Toast.makeText(getContext(), error.toString().trim(), Toast.LENGTH_SHORT).show()){
            @Override
            protected Map<String, String> getParams() throws AuthFailureError {
                Map<String, String> data = new HashMap<>();

                data.put("username", username);
                data.put("phone", phone);
                data.put("password", pass);
                data.put("identity", "0");
                data.put("remarks_ill", remarks_ill);
                data.put("remarks_eating", remarks_eating);
                data.put("remarks_other", remarks_other);
                data.put("contactPhone", contactPhone);
                data.put("contactEmail", contactEmail);
                Log.d("connect", remarks_eating);
                return data;
            }
        };
        RequestQueue requestQueue = Volley.newRequestQueue(getContext());
        requestQueue.add(stringRequest);
    }


}

File: app/src/main/java/com/example/elderapp/register/RegisterElderFragment.kt
package com.example.elderapp.register

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.fragment.app.Fragment
import com.android.volley.AuthFailureError
import com.android.volley.Response
import com.android.volley.VolleyError
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import com.example.elderapp.Global
import com.example.elderapp.LoginActivity
import com.example.elderapp.R
import java.util.*

class RegisterElderFragment : Fragment() {
    private var etName: EditText? = null
    private var etPhone: EditText? = null
    private var etPassword: EditText? = null
    private var etPasswordC: EditText? = null
    private var etRemarksIll: EditText? = null
    private var etRemarksEating: EditText? = null
    private var etRemarksOther: EditText? = null
    private var etContactPhone: EditText? = null
    private var etContactEmail: EditText? = null
    private var etAddress: EditText? = null
    private var username: String? = null
    private var phone: String? = null
    private var pass: String? = null
    private var passC: String? = null
    private var remarksIll: String? = ""
    private var remarksEating: String? = ""
    private var remarksOther: String? = ""
    private var contactPhone: String? = null
    private var contactEmail: String? = null
    private var sex: String? = null
    private var addr: String? = null
    private val url: String? = "https://www2.cs.ccu.edu.tw/~lwx109u/elderApp/register.php"
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val root = inflater.inflate(R.layout.fragment_register_elder, container, false)
        etName = root.findViewById(R.id.et_name)
        etPhone = root.findViewById(R.id.et_phone)
        etPassword = root.findViewById(R.id.et_password)
        etPasswordC = root.findViewById(R.id.et_passwordCheck)
        etRemarksIll = root.findViewById(R.id.et_remarks1)
        etRemarksEating = root.findViewById(R.id.et_remarks2)
        etRemarksOther = root.findViewById(R.id.et_remarks3)
        etContactPhone = root.findViewById(R.id.et_contactPhone)
        etContactEmail = root.findViewById(R.id.et_contactEmail)
        etAddress = root.findViewById(R.id.et_address)
        val register = root.findViewById<Button?>(R.id.btn_register)
        val toLogin = root.findViewById<TextView?>(R.id.tv_toLogin)
        val radioGroup = root.findViewById<RadioGroup>(R.id.RadioGroup_sex)

        register.setOnClickListener {
            username = etName!!.text.toString().trim { it <= ' ' }
            phone = etPhone!!.text.toString().trim { it <= ' ' }
            pass = etPassword!!.text.toString().trim { it <= ' ' }
            passC = etPasswordC!!.text.toString().trim { it <= ' ' }
            remarksIll = etRemarksIll!!.text.toString().trim { it <= ' ' }
            remarksEating = etRemarksEating!!.text.toString().trim { it <= ' ' }
            remarksOther = etRemarksOther!!.text.toString().trim { it <= ' ' }
            contactPhone = etContactPhone!!.text.toString().trim { it <= ' ' }
            contactEmail = etContactEmail!!.text.toString().trim { it <= ' ' }

            addr = etAddress!!.text.toString().trim()
            when(radioGroup.checkedRadioButtonId){
                R.id.RadioButton_M -> sex = "M"
                R.id.RadioButton_F -> sex = "F"
                R.id.RadioButton_N -> sex = "N"
            }

            if (pass != passC) {
                Global.putSnackBarR(etName!!, "密碼不相符")
            } else if (username == "" || phone == "" || contactPhone == "") {
                Global.putSnackBarR(etName!!, "請輸入完整資訊")
            } else {
                SQL()
            }
        }
        toLogin.setOnClickListener { startActivity(Intent(context, LoginActivity::class.java)) }
        return root
    }

    fun SQL() {
        val stringRequest: StringRequest = object : StringRequest(Method.POST, url, Response.Listener { response: String ->
            Log.d("connect", "Response: $response")
            if (response.startsWith("success")) {
                val it = Intent(context, LoginActivity::class.java)
                it.putExtra("signUp", true)
                startActivity(it)
                activity?.finish()
            } else if (response.startsWith("failure")) {
                Toast.makeText(context, "註冊失敗", Toast.LENGTH_SHORT).show()
            }
        }, Response.ErrorListener { error: VolleyError -> Toast.makeText(context, error.toString(), Toast.LENGTH_SHORT).show() }) {
            @Throws(AuthFailureError::class)
            override fun getParams(): MutableMap<String?, String?> {
                val data: MutableMap<String?, String?> = HashMap()
                data["username"] = username
                data["phone"] = phone
                data["password"] = pass
                data["identity"] = "0"
                data["remarks_ill"] = remarksIll
                data["remarks_eating"] = remarksEating
                data["remarks_other"] = remarksOther
                data["contactPhone"] = contactPhone
                data["contactEmail"] = contactEmail
                data["address"] = addr
                data["sex"] = sex

                return data
            }
        }
        val requestQueue = Volley.newRequestQueue(context)
        requestQueue.add(stringRequest)
    }
}